odf/opendocument.py

```diff
@@ -568,8 +568,11 @@ def save(self, outputfile, addsuffix=False):
         the ZIP content.
         @param addsuffix boolean: whether to add a suffix or not; defaults to False
         """
-        assert(type(outputfile)==type(u"") or 'wb' in repr(outputfile) or 'BufferedWriter' in repr(outputfile)  or 'BytesIO' in repr(outputfile))
```

Comment:
I don't think they are useful.  The python way would be to pass odfFile to zipfile.ZipFile(), and let it fail if the argument is invalid.

I was just being conservative, and fixing a bug where string (non-unicode) was not accepted, and use instanceof(x, type) instead of "type" in repr(x).